### PR TITLE
Get rid of trolls opening inventories!

### PR DESCRIPTION
**Pull request by Mane48**
## Description

Hey, yes. Literally, it will try its best to get rid of the trolls that open the inventory. If they are modding and try the famous [QB] Open Inventory trigger it will yeet them from the server and ban them!

_Im just waiting on a modder to join my server and try it so I can see it work!_ 

It does work! I kinda see how it does up the amount of code that is required but ay... it'll def reduce the amount of trolls!

Update :

Upon more testing i've realized that some of staff may be banned, its not an always kind of deal but there is a possibility, I dont know what exactly could cause this but if anyone has any suggestions feel free to push them!

Update 2 :

Downside is that it doesnt actually work properly if they're close to you. It'll just run it like either a robbery or so.

## Checklist

- [x] I have personally loaded this code into an updated qbcore project and checked all of its functionality.
- [x] My code fits the style guidelines.
- [x] My PR fits the contribution guidelines.

### CODE REVIEW

File: server/main.lua [hunk 1]
```diff
@@ -10,7 +10,19 @@ local ShopItems = {}
 --#endregion Variables
 
 --#region Functions
-
+local function BanPlayer(src)
```
Comment (Sharkiiie) [inline server/main.lua:13]: I would suggest using the ExploitBan export from core to avoid redundant code.

File: server/main.lua [hunk 2]
```diff
@@ -1223,17 +1235,59 @@ local function OpenInventory(name, id, other, origin)
...
+                    end
+				end
+			else
+				local invisible = IsEntityVisible(PlayerPed)
+				local invincible = GetPlayerInvincible(PlayerPed)
+				
+				if invisible == false or invincible == true then
+					if not Perms.admin or not Perms.god then
+						TriggerEvent("qb-log:server:CreateLog", "anticheat", "qb-inventory", "red", string.format("User: ** %s **\nIdentifier: ** %s **\nCitizenId: ** %s **\nServer Id: ** %s ** Was using either invisibility or noclip while trying to open an inventory", GetPlayerName(src), GetPlayerIdentifier(src), Player.PlayerData.citizenid, src))
+						DropPlayer(src, "Invisible or invincible? Which one 😂")
```
Comment (Sharkiiie) [inline server/main.lua:1272]: Player drop messages should be clear and concise, along the lines of "you have been kicked / banned for x"
I would suggest putting the drop message in the locales files as well.

File: server/main.lua [hunk 3]
```diff
@@ -1595,18 +1649,59 @@ RegisterNetEvent('inventory:server:OpenInventory', function(name, id, other)
...
+                        Wait(250)
+                    end
+				end
+			else
+				local invisible = IsEntityVisible(PlayerPed)
+				local invincible = GetPlayerInvincible(PlayerPed)
+				if invisible == false or invincible == true then
+					if not Perms.admin or not Perms.god then
+						TriggerEvent("qb-log:server:CreateLog", "anticheat", "qb-inventory", "red", string.format("User: ** %s **\nIdentifier: ** %s **\nCitizenId: ** %s **\nServer Id: ** %s ** Was using either invisibility or noclip while trying to open an inventory", GetPlayerName(src), GetPlayerIdentifier(src), Player.PlayerData.citizenid, src))
+						DropPlayer(src, "Invisible or invincible? Which one 😂")
```
Comment (Sharkiiie) [inline server/main.lua:1686]: Player drop messages should be clear and concise, along the lines of "you have been kicked / banned for x"
I would suggest putting the drop message in the locales files as well.

**Review by Sharkiiie — CHANGES_REQUESTED** [review]
Follow linting and styling guidelines.
Please make sure the file is using 4 spaces per indent rather than tabs, and each new block of code is indented one time.

I would highly suggest using the Lua extension by sumneko and enabling the formatter.
It makes it much easier to follow the guidelines and does most of the work for you.

File: server/main.lua (hunk 3 above)
Comment (Mane48) [reply server/main.lua:1686]: Alright sure thing, I appreciate it ill implement and fix all the things you mentioned!

**Comment by Sharkiiie** [comment]
Still not following linting / styling guidelines for spaces rather than tabs.
I'll have to give everything else a closer review later.

**Comment by Mane48** [comment]
Hey I thought I actually did follow it. In my side they show up as spaces 4 spaces instead of tabs. How can I check this again?

**Pull request closed without merging**